src/utils/server-utils.ts

```diff
@@ -225,6 +231,16 @@ async function getPointDataCoverage(layer: PointDataLayerProps) {
   return possibleDates;
 }
 
+async function getNSOAvailableDates(layer: NSOLayerProps) {
```

Comment:
We should add a comment or rename this function in such a way that other devs won't get confused and use this to get the dates instead of `getPossibleDatesForLayer`, since by how its framed, it could be used independently. 

Maybe a comment highlighting the data this outputs is being saved in the server state slice as `availableDates`?